### DISCUSSION TITLE
How to Exclude Specific <Page> Items from WASM Build using Uno.Sdk Single Project?

### DISCUSSION THREAD

**Discussion by AdriaanLarcai** [unoplatform/uno#18866] [Q&A] [answered]
Hello Uno Platform community,

I'm currently working on a project using Uno Platform with the new `Uno.Sdk` single project template. I am targeting multiple platforms, including WebAssembly (WASM). I've encountered an issue where all `<Page>` items in my project seem to be included in the WASM build by default.

Despite my efforts to set `EnableDefaultItems` to `false` in the `.csproj` file, it appears that the `<Page>` items are still automatically included, and I am unable to exclude specific pages from the WASM build. I've also tried manually marking the page item to be removed using `<Page Remove="Page.xaml"/>`, but this approach does not seem to work either. Here's a snippet of what I've tried:

```xml
<Project Sdk="Uno.Sdk">
    <PropertyGroup>
        <TargetFrameworks>net9.0-windows10.0.26100;net9.0-browserwasm</TargetFrameworks>
        <UnoSingleProject>true</UnoSingleProject>
        <OutputType>Library</OutputType>
        <!-- Ensures the .xr.xml files are generated in a proper layout folder -->
        <GenerateLibraryLayout>true</GenerateLibraryLayout>
        <ImplicitUsings>disable</ImplicitUsings>
        <Nullable>disable</Nullable>
        <DefaultLanguage>en-US</DefaultLanguage>
        <EnableDefaultItems>False</EnableDefaultItems>
        <!--
      UnoFeatures let's you quickly add and manage implicit package references based on the features you want to use.
      https://aka.platform.uno/singleproject-features
    -->
        <UnoFeatures>
            Hosting;
            Logging;
            Mvvm;
        </UnoFeatures>
        <Platforms>AnyCPU</Platforms>
    </PropertyGroup>
    <ItemGroup Condition="'$(TargetFramework)' == 'net9.0-browserwasm'">
        <None Remove="Page.xaml"/>
        <Page Remove="Page.xaml"/>
    </ItemGroup>
</Project>
```

**What I Want to Achieve:**

I would like to selectively exclude certain `<Page>` items from being included in the WASM build while keeping them for other target platforms.

**Questions:**

1. Is there a specific way to exclude certain `<Page>` items from the WASM build when using the `Uno.Sdk` single project template?
2. Are there any recommended practices or workarounds for handling platform-specific page inclusion/exclusion in a multi-targeted Uno Platform project?
3. Could there be any project configuration or tooling updates that address this requirement?

Any guidance or examples from the community would be greatly appreciated!

Thank you!

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the question!

Given the structure of files inclusion in the Uno.SDK, the only way to remove files is to remove the files in a target, like this:

```xml
<Target Name="AdjustAppItemGroups" BeforeTargets="ResolveAssemblyReferences">
    <ItemGroup Condition="'$(TargetFramework)' == 'net9.0-browserwasm'">
        <None Remove="Page.xaml"/>
        <Page Remove="Page.xaml"/>
    </ItemGroup>
</Target>
```

This target is executed as part of Design-Time builds and should be applied at the right times.

Give it a try and let us know!

**Reply by MartinZikmund** [reply]
Could be useful to add this to docs @morning4coffe-dev
